src/audio/mux/mux.c

```diff
@@ -629,6 +629,11 @@ static int mux_reset(struct comp_dev *dev)
 	comp_info(dev, "mux_reset()");
 
 	if (dir == SOF_IPC_STREAM_PLAYBACK) {
+		/*
+		 * On playback path, stop propagating reset signal to downstream if any of the
+		 * source is active. The condition is only applicable to comp MUX (multi-source to
+		 * single-sink).
+		 */
```

Comment:
There's no explicit check for the mux / demux type here. We just implicitly use the fact, that in case of a demux we would be coming into reset from the upstream source direction, so the single source is guaranteed to already be inactive? So the below condition `if (state > COMP_STATE_READY)` can only trigger for a mux?

Reply:
AFAIK reset trigger starts from PCM and propagates downstream/upstream for playback/capture. It implies that for DEMUX on playback, reset can be received only when its (single) source got reset. Same concept for MUX on capture.

However I'm not sure if the trigger/propagation rule will be changed in the future. For example of MUX on capture, would it be possible to reset a PCM-less pipeline which presents as a branch of MUX upstream in multiple endpoints scheme? (like PPL#3 in the following graph)
```
PCMC --- PPL#1 ---> MUX --- PPL#2 ---> DAI#2
                       \--> PPL#3 ---> DAI#3